truss-chains/truss_chains/framework.py

```diff
@@ -614,73 +557,75 @@ def __init__(
     """
 
     _location: _ErrorLocation
-    has_context: bool
+    cls: Type[definitions.ABCChainlet]
+    has_context: bool = False
     validated_dependencies: Mapping[str, definitions.DependencyDescriptor]
 
     def __init__(
         self, cls: Type[definitions.ABCChainlet], location: _ErrorLocation
     ) -> None:
-        if not cls.has_custom_init():
+        self.cls = cls
+        if not self.cls.has_custom_init():
             self.has_context = False
             self.validated_dependencies = {}
             return
-        # Each validation pops of "processed" arguments from the list.
         line = inspect.getsourcelines(cls.__init__)[1]
         self._location = location.model_copy(
             update={"line": line, "method_name": "__init__"}
         )
+
         params = list(inspect.signature(cls.__init__).parameters.values())
-        params = self._validate_self_arg(list(params))
-        params, self.has_context = self._validate_context_arg(params)
-        self.validated_dependencies = self._validate_dependencies(params)
+        self._validate_args(params)
 
-    def _validate_self_arg(
-        self, params: list[inspect.Parameter]
-    ) -> list[inspect.Parameter]:
+    def _validate_args(self, params: list[inspect.Parameter]):
+        # Each validation pops of "processed" arguments from the list.
+        self._validate_self_arg(params)
```

Comment:
I consolidated all the `_validate` functions here with a slight refactor. I thought it was a little confusing that the functions will modify input (clued by comment) but the former code still reassigned the `params` variable. 

My slight preference would be to include the comment, but write the code so that it's more clear that the `params` variable is getting modified in place